stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketObservability.java

```diff
@@ -0,0 +1,259 @@
+    public static final String WEBSOCKET_MESSAGE_TYPE_CLOSE = "close";
+
+    static final String WEBSOCKET_ERROR_TYPE_CONNECTION = "connection";
+    public static final String WEBSOCKET_ERROR_TYPE_CLOSE = "close";
+    public static final String WEBSOCKET_ERROR_TYPE_MESSAGE_SENT = "message_sent";
+    static final String WEBSOCKET_ERROR_TYPE_MESSAGE_RECEIVED = "message_received";
+
+    //Observe all WebSocket connection requests
+    static void observeRequest(WebSocketOpenConnectionInfo connectionInfo, String result) {
+        if (ObserveUtils.isObservabilityEnabled()) {
+            ObserverContext observerContext = new ObserverContext();
+            observerContext.setConnectorName(SERVER_CONNECTOR_WEBSOCKET);
+            try {
+                observerContext.addTag(TAG_CONNECTION_ID, connectionInfo.getWebSocketConnection().getChannelId());
+            } catch (IllegalAccessException e) {
+                //TODO: Handle Exception
+            }
+
+            setObserveService(observerContext, connectionInfo);
+
+            observerContext.addTag(TAG_KEY_RESULT, result);
+            Map<String, String> tags = observerContext.getTags();
+            Set<Tag> allTags = new HashSet<>(tags.size());
+            Tags.tags(allTags, observerContext.getTags());
+
+            MetricRegistry metricRegistry = DefaultMetricRegistry.getInstance();
+
+            //Increment requests metric
+            metricRegistry.counter(new MetricId(SERVER_CONNECTOR_WEBSOCKET + "_" + METRIC_REQUESTS,
+                                                METRIC_REQUESTS_DESC, allTags)).increment();
+        }
+    }
+
+    //Observe all WebSocket connections
+    static void observeConnection(WebSocketOpenConnectionInfo connectionInfo) {
+        if (ObserveUtils.isObservabilityEnabled()) {
+
+            ObserverContext observerContext = new ObserverContext();
+            observerContext.setConnectorName(SERVER_CONNECTOR_WEBSOCKET);
+
+            try {
+                observerContext.addTag(TAG_CONNECTION_ID,
+                                       connectionInfo.getWebSocketConnection().getChannelId());
+            } catch (IllegalAccessException e) {
+                //TODO: handle exception
+            }
+            setObserveService(observerContext, connectionInfo);
+
+            Map<String, String> tags = observerContext.getTags();
+            Set<Tag> allTags = new HashSet<>(tags.size());
+            Tags.tags(allTags, observerContext.getTags());
+
+            MetricRegistry metricRegistry = DefaultMetricRegistry.getInstance();
+
+            //Increment current connections metric
+            metricRegistry.gauge(new MetricId(SERVER_CONNECTOR_WEBSOCKET + "_" + METRIC_CONNECTIONS,
+                                              METRIC_CONNECTIONS_DESC, allTags)).increment();
+
+        }
+    }
+
+    //Observe all messages sent (pushed)
+    public static void observePush(String type, String result, WebSocketOpenConnectionInfo connectionInfo) {
```

Comment:
Shall we use proper java doc comments here?

Reply:
added